lib/screen_checker/vendor_data/fetch.ex

```diff
@@ -0,0 +1,40 @@
+defmodule ScreenChecker.VendorData.Fetch do
+  @moduledoc false
+
+  require Logger
+
+  def make_and_parse_request(url, headers \\ [], parse_fn, vendor_name, opts) do
+    with {:http_request, {:ok, response}} <- {:http_request, HTTPoison.get(url, headers, opts)},
+         {:response_success, %{status_code: 200, body: body}} <- {:response_success, response},
+         {:parse, {:ok, parsed}} <- {:parse, parse_fn.(body)} do
+      {:ok, parsed}
+    else
+      {:http_request, {:error, e}} ->
+        log_fetch_error(vendor_name, :http_fetch_error, %{message: HTTPoison.Error.message(e)})
+
+      {:response_success, %{status_code: status_code}} ->
+        log_fetch_error(vendor_name, :bad_response_code, %{status_code: status_code})
+
+      {:parse, _} ->
+        log_fetch_error(vendor_name, :parse_error)
+
+      _ ->
+        log_fetch_error(vendor_name, :error)
+    end
+  end
+
+  defp log_fetch_error(vendor_name, e) do
+    _ = Logger.info("#{vendor_name}_fetch_error #{e}")
```

Comment:
is there any particular reason for the `_` matching in some of these function calls? See it in a few spots and not sure if it's a decided upon convention vs an unmatched function call. Mostly just curious.

Reply:
This might not be necessary anymore, but in the past Credo would yell at us when we didn't use the return value of a function call that wasn't very obviously for side-effects only (e.g. `IO.puts`, `Logger.info`). I think it would be ok without the `_ =` on this one instance, but yeah in general it's a best-practices thing to say "this function is being called solely for its side-effects".